script.js

```diff
@@ -1,4 +1,235 @@
+
+  console.log("rollDice output, randomInteger: ", randomInteger);
+  return randomInteger;
+};
+
+// helper function to roll dice for player
+var rollDiceForPlayer = function () {
+  console.log(`Control flow: start of rollDiceForPlayer()`);
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRolls.push(rollDice());
+    counter = counter + 1;
+  }
+
+  console.log(`rollDiceForPlayer changes, playerRolls: `, currentPlayerRolls);
+  return `ROUND ${roundCounter}!<br><br>
+  Welcome, Player ${currentPlayer}. <br> <br> You rolled: <br> Dice 1: ${currentPlayerRolls[0]} | Dice 2: ${currentPlayerRolls[1]} <br><br> Now, please input either "1" or "2" to choose corresponding dice to be used as the first digit of your final value.</br>`;
+};
+
+var getPlayerScore = function (playerInput) {
+  var playerScore;
+  // input validation
+  if (!(playerInput == 1 || playerInput == 2)) {
+    return `Invalid input. Game will not continue until you enter either "1" or "2".<br><br>
+    You rolled: <br> Dice 1: ${currentPlayerRolls[0]} | Dice 2: ${currentPlayerRolls[1]}`;
+  }
+  // input == 1
+  else if (playerInput == 1) {
+    console.log(`Control flow: input == 1`);
+    playerScore = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+  }
+  // input == 2
+  else if (playerInput == 2) {
+    console.log(`Control flow: input == 2`);
+    playerScore = Number(
+      String(currentPlayerRolls[1]) + String(currentPlayerRolls[0])
+    );
+  }
+
+  // Store playerScore in array
+  allPlayersScore.push(playerScore);
+
+  // store overall player score until browser is refreshed
+  overallPlayersScore.push(playerScore);
+
+  // clear current player rolls array
+  currentPlayerRolls = [];
+  return `Player ${currentPlayer}, your chosen value is: ${playerScore}`;
+};
+
+var comparePlayersScores = function () {
+  var compareMessage = `Player 1 score: ${allPlayersScore[0]} <br> Player 2 score: ${allPlayersScore[1]}`;
+  // player 1 wins
+  if (allPlayersScore[0] > allPlayersScore[1]) {
+    compareMessage = `${compareMessage} <br><br> Player 1 wins!`;
+  }
+  // player 2 wins
+  if (allPlayersScore[0] < allPlayersScore[1]) {
+    compareMessage = `${compareMessage} <br><br> Player 2 wins!`;
+  }
+  // tie
+  if (allPlayersScore[0] == allPlayersScore[1]) {
+    compareMessage = `${compareMessage} <br><br> It's a tie!`;
+  }
+  return compareMessage;
+};
+
+// need a function that will calculate the sum of numbers based on when/where they were pushed in the array
+var runningSumOfNumbers = function () {
+  console.log(`adding player scores`);
+  while (i < overallPlayersScore.length) {
+    runningPlayerScore1 += overallPlayersScore[i];
+    console.log(`running player 1 score is: ${runningPlayerScore1}`);
+    i += 2;
+  }
+  while (j < overallPlayersScore.length) {
+    runningPlayerScore2 += overallPlayersScore[j];
+    console.log(`running player 2 score is: ${runningPlayerScore2}`);
+    j += 2;
+  }
+};
```

Comment:
this feels like a somewhat roundabout way of calculating the running sum of numbers, also if you're using i/j as an index here for a loop, it's ALWAYS best to keep the variable here, and not make it a global scope.

Now rather than having 2 separate loops here, we can make it more succinct with one loop through. As if let's say `overallPlayersScore.length` is something VERY big, it will take very long to complete 2 loops as opposed to just one.

We can probably make the loop like so:
```
for(var i = 0; i<overallPlayersScore.length; i+=1){
  if(i%2 == 0){
    runningPlayerScore1 += overallPlayersScore[i]
  } else {
    runningPlayerScore2 += overallPlayersScore[i]
  }
}
```

this way it's only one loop through, and we did all the things!